ui/src/app/settings/actions/config.js

```diff
@@ -13,4 +13,18 @@ config.fetch = () => {
   };
 };
 
+config.saveConfig = (name, value) => {
+  return {
+    type: "WEBSOCKET_SEND",
+    payload: {
+      actionType: "SAVE_CONFIG",
+      message: {
+        method: "config.update",
+        params: { name, value },
```

Comment:
To answer your first question TODO question, fairly certain this can be an array.

Reply:
@squidsoup Hmm yeah I wasn't sure about that. This is the update method in the config websocket handler, which seems to expect a dict with name and value keys:

``` python
    def update(self, params):
        """Update a config value."""
        if not self.user.is_superuser:
            raise HandlerPermissionError()
        if 'name' not in params:
            raise HandlerPKError("Missing name in params")
        if 'value' not in params:
            raise HandlerValidationError("Missing value in params")
        name = params['name']
        value = params['value']
        form = get_config_form(name, {name: value})
        if form.is_valid():
            try:
                request = HttpRequest()
                request.user = self.user
                form.save(ENDPOINT.UI, request)
            except ValidationError as e:
                self._fix_validation_error(name, e.error_dict)
                raise HandlerValidationError(e.error_dict)
            return self._include_choice({
                'name': name,
                'value': Config.objects.get_config(name),
            })
        else:
            self._fix_validation_error(name, form.errors)
            raise HandlerValidationError(form.errors)
```

Unless I'm missing something (highly possible)?

Reply:
Yep, the API is a bit broken here, but we should not be dispatching multiple actions, the payload should still be an array - we'll just have to handle that in the saga.

Reply:
Have filed [a bug](https://bugs.launchpad.net/maas/+bug/1839788).

Reply:
@Caleb-Ellis support for [params as an array landed](https://github.com/canonical-web-and-design/maas-ui/commit/d71059c4b9709ae0b50c91c4cb7428322f43fb0e).